xapian-core/weight/tfidfweight.cc

```diff
@@ -211,42 +372,44 @@ TfIdfWeight::get_wdfn(Xapian::termcount wdf, Xapian::termcount doclen,
 	    return num / den;
 	}
 	default:
-	    AssertEq(c, 'n');
+	    AssertEq(wdf_norm_, WDF_NORM::NONE);
 	    return wdf;
     }
 }
 
 double
-TfIdfWeight::get_idfn(char c) const
+TfIdfWeight::get_idfn(IDF_NORM idf_norm_) const
 {
     Xapian::doccount termfreq = 1;
-    if (c != 'n') termfreq = get_termfreq();
+    if (idf_norm_ != IDF_NORM::NONE) termfreq = get_termfreq();
     double N = 1.0;
-    if (c != 'n' && c != 'f') N = get_collection_size();
-    switch (c) {
-	case 'n':
+    if ((idf_norm_ == IDF_NORM::PROB) || (idf_norm_ == IDF_NORM::SQUARE) ||
+	(idf_norm_ == IDF_NORM::TFIDF) || (idf_norm_ == IDF_NORM::PIVOTED))
+	N = get_collection_size();
+    switch (idf_norm_) {
+	case IDF_NORM::NONE:
 	    return 1.0;
-	case 'p':
+	case IDF_NORM::PROB:
 	    // All documents are indexed by the term
 	    if (N == termfreq) return 0;
 	    return log((N - termfreq) / termfreq);
-	case 'f':
+	case IDF_NORM::FREQ:
 	    return (1.0 / termfreq);
-	case 's':
+	case IDF_NORM::SQUARE:
 	    return pow(log(N / termfreq), 2.0);
-	case 'P':
+	case IDF_NORM::PIVOTED:
 	    return log((N + 1) / termfreq);
 	default:
-	    AssertEq(c, 't');
+	    AssertEq(idf_norm_, IDF_NORM::TFIDF);
 	    return (log(N / termfreq));
     }
 }
 
 double
-TfIdfWeight::get_wtn(double wt, char c) const
+TfIdfWeight::get_wtn(double wt, WT_NORM wt_norm_) const
 {
-    (void)c;
-    AssertEq(c, 'n');
+    (void)wt_norm_;
+    AssertEq(wt_norm_, WT_NORM::NONE);
```

Comment:
Since this is now an enum, you don't need to assert anything — the compiler should pick up any attempt to violate this.

Reply:
Updated